docs/architecture.md

```diff
@@ -29,73 +29,100 @@ have at least 3 nodes. In a basic setup with this amount of nodes,
-respectively).
-
-The Operator functionality extends the Kubernetes API with
-*PerconaServerMySQL* object, and it is implemented as a golang
-application. Each *PerconaServerMySQL* object maps to one separate MySQL setup.
-The Operator listens to all events on the created objects.
-When a new PerconaServerMySQL object is created, or an existing one undergoes
-some changes or deletion, the Operator automatically
-creates/changes/deletes all needed Kubernetes objects with the
-appropriate settings to provide a proper Percona Server for MySQL operation.
+With Group
+Replication, recovery is coordinated by the group membership protocol. 
+
+![image](assets/images/operator-GR.svg)
+
+Clients connect
+via HAProxy or MySQL Router to reach a healthy writer/reader instance.
+
+## How the Operator works
+
+The Operator extends the Kubernetes API with a special `PerconaServerMySQL` object. It's a Go application that watches for events on these objects. Each `PerconaServerMySQL` object maps to one separate MySQL setup.
+
+When you create, change, or delete a `PerconaServerMySQL` object, the Operator automatically creates or updates the necessary Kubernetes resources, like Pods and StatefulSets, to ensure your database cluster matches the desired state.
+
+For storage, the Operator leverages Persistent Volumes (PVs) and PersistentVolumeClaims (PVCs) to ensure your data is durable. A *PersistentVolumeClaim* (PVC) is used to implement
+the automatic storage provisioning to pods. The Container Storage Interface (CSI) allows the storage to be re-mounted on a different node if a failure occurs.
 
 ## Replication types and proxy solutions
 
 The process in which data from a *primary* MySQL instance is copied and sent to
-other MySQL instances of the database cluster (co-called *replicas*) is known as
+other MySQL instances of the database cluster (so-called *replicas*) is known as
 *MySQL replication*.
 
 The Operator [provides you a choice](operator.md#mysqlclustertype) between two
-replication types: 
+replication types:
 
-* **Asynchronous replication** means that write is completed on the
+* **Asynchronous replication** means that a write is completed on the
     primary and is not influenced by the replicas themselves. After completion
     of its changes, the primary populates the binary log with the data
     modification or the actual statement. Then the replica executes each change
     on its own database and obtains consistent set of data.
 
-* **Group Replication** means that read-write transaction can be executed on any
+* **Group Replication** means that a read-write transaction can be executed on any
     instance, and when it happens, MySQL tries to get consensus with the other
-    instances before returning it completed back to the client. 
+    instances before returning it completed back to the client.
 
 Both replication types have their pros and cons.
 
 *Asynchronous replication* is faster. Also, you can distribute read and write
```

Comment:
Nope, you can distribute only READS, and with the high chance to get stale reads given the data is asynchronously updated.